File: solved/141A.rb
def check_pile(sa, sb, lf)
  freq_ab = Hash.new(0)
  sa.each_char {|c| freq_ab[c] += 1}
  sb.each_char {|c| freq_ab[c] += 1}

  # If any of the letters counted from the words is greater or less than 0,
  # it means that there are missing chars or more than expected
  # 0< Non existent characters
  # 0< More characters than expected
  # 0> Missing characters
  lf.each_char {|c| freq_ab[c] -= 1}
  freq_ab.each_value do |v|
    if v == 0
      next
    else
      return "NO"
    end
  end
  "YES"
end

string_a = gets.chomp
string_b = gets.chomp
letters_found = gets.chomp

puts check_pile(string_a, string_b, letters_found)

File: solved/141A.cc
#include <iostream>
#include <vector>
#include <map>

using namespace std;

int main() {
  map<char, int> letters;
  string t;
  int total = 0;

  cin >> t;
  int tl = t.length();
  for(int i=0; i<tl; ++i) {
    if(letters.find(t[i]) == letters.end())
      letters[t[i]] = 1;
    else
      letters[t[i]]++;
    ++total;
  }

  cin >> t;
  tl = t.length();
  for(int i=0; i<tl; ++i) {
    if(letters.find(t[i]) == letters.end())
      letters[t[i]] = 1;
    else
      letters[t[i]]++;
    ++total;
  }

  cin >> t;
  tl = t.length();
  for(int i=0; i<tl; ++i)
    if(letters.find(t[i]) == letters.end()) {
      cout << "NO";
      return 0;
    } else {
      letters[t[i]]--;
      total--;
      if(letters[t[i]] < 0) {
        cout << "NO";
        return 0;
      }
    }

  cout << (total == 0 ? "YES" : "NO");
  return 0;
}
